ckan/public/base/javascript/modules/autocomplete.js

```diff
@@ -186,15 +186,20 @@ this.ckan.module('autocomplete', function (jQuery) {
      *
      * Returns a text string.
      */
-    formatResult: function (state, container, query) {
-      var term = this._lastTerm || null; // same as query.term
+    formatResult: function (state, container, query, escapeMarkup) {
+      var term = this._lastTerm || (query ? query.term : null) || null; // same as query.term
 
       if (container) {
         // Append the select id to the element for styling.
         container.attr('data-value', state.id);
       }
 
-      return state.text.split(term).join(term && term.bold());
+      var result = [];
+      $(state.text.split(term)).each(function() {
+        result.push(escapeMarkup ? escapeMarkup(this) : this);
+      });
+
+      return result.join(term && (escapeMarkup ? escapeMarkup(term) : term).bold());
```

Comment:
With the above comment in mind, you could probably be able to get away with removing lines 197 to 200, because its escaping empty strings. And then changing line 202 to something like `return state.text.split(term).join(term && (escapeMarkup ? escapeMarkup(term) : term).bold());`

But test before making changes, i've looked at it too long lol

Reply:
In my opinion, xss issue will not be resolved without escaping state.text.